internal/config/config.go

```diff
@@ -132,14 +142,45 @@ func Load(path string, opts *LoadOptions) (*Config, error) {
 	}
 
 	return &Config{
-		hclConfig: cfg,
-		ctx:       ctx,
-		path:      filepath.Dir(path),
-		pathData:  pathData,
+		hclConfig:      cfg,
+		ctx:            ctx,
+		path:           filepath.Dir(path),
+		pathData:       pathData,
+		InputVariables: vs,
 	}, nil
 }
 
 // HCLContext returns the eval context for this configuration.
 func (c *Config) HCLContext() *hcl.EvalContext {
 	return c.ctx.NewChild()
 }
+
+func DecodeVariableBlocks(body hcl.Body) (map[string]*variables.Variable, hcl.Diagnostics) {
```

Comment:
This needs a doc comment.

Reply:
Also, I don't think this needs to be exported? I can't see anything else calling it.

Reply:
Also, should this be in the `variables` package? (not rherotical, real question)

Reply:
Ah I think I forgot to unexport it when I moved it from `variables` to `config`. To your question, my thinking was that this is a thing that `config` is doing to decode things in the `config`, and then this `config` function calls the necessary `variables` function.
But I don't have strong feelings about it, so if in go-isms it makes more sense to put it in `variables` I'm down with that.

Reply:
Given your other comment, I'll go ahead and move this into `variables`.